builds/delete-serverless-service-build-pack/Jenkinsfile

```diff
@@ -255,6 +259,42 @@ node {
 
 }
 
+/**
+ * Check bucket existance and content in it. if yes then empty it for SLS delete
+ * @param  stage
+ * @return
+ */
+def checkBucket(stage) {
+	withCredentials([
+		[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']
+	]) {
+		try {
+			sh "aws configure set profile.cloud-api.region ${configLoader.AWS.REGION}"
+			sh "aws configure set profile.cloud-api.aws_access_key_id $AWS_ACCESS_KEY_ID"
+			sh "aws configure set profile.cloud-api.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"					
+			
+			// Empty bucket
+			def targetBucket = service_config['event_source_s3'];
+			if(stage != "prod"){
+				targetBucket = service_config['event_source_s3'] + "-" + stage;
+			}
+			def s3Exists = isBucketEmpty(targetBucket);											
```

Comment:
isBucketEmpty as  the name says, it will check whether the bucket empty or not. First check bucket exists or not and then bucket empty. We can take this offline, if you have need any clarification

Reply:
this logic is wrong. is BucketEmpty is giving you the value of false if there are contents in the bucket. You are then using that assuming it will be true to run the logic to delete. But what you are getting is value of false. You need to change the variable name to say something like s3Empty and then check for if (!s3Empty)